src/Bicep.Core/Analyzers/Linter/Rules/SecureParamsInNestedDeploymentsRule.cs

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Navigation;
+using Bicep.Core.Semantics;
+using Bicep.Core.Semantics.Namespaces;
+using Bicep.Core.Syntax;
+using Bicep.Core.Syntax.Visitors;
+using Bicep.Core.TypeSystem;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class SecureParamsInNestedDeploymentsRule : LinterRuleBase
+    {
+        // Finds insecure uses of parameters or list* functions inside outer-scpped nested deployments (resources of
+        //   type "microsoft.resources/deployments").  Modules are inner-scoped so do not have this issue.
+        //
+        // These nested deployments will evaluate expressions prior to being sent to the deployment engine.This means that all
+        // properties are in the request in an evaluated state (clear text) and persisted on the deployment object.
+
+        public new const string Code = "secure-params-in-nested-deploy";
+
+        public SecureParamsInNestedDeploymentsRule() : base(
+            code: Code,
+            description: CoreResources.SecureParamsInNestedDeployRule_Description,
+            docUri: new Uri($"https://aka.ms/bicep/linter/{Code}"),
+            diagnosticLevel: DiagnosticLevel.Warning)
```

Comment:
Default is warning. So we can skip this.

Reply:
Probably going to make it required when I do the new levels.  I want people to think about it and code reviewers to see it explicityly.